From 4da3eb136d660d27be12490fc11105fea3f319ab Mon Sep 17 00:00:00 2001
From: Andy Li <andy@onthewings.net>
Date: Thu, 12 May 2016 16:41:32 +0800
Subject: [PATCH] run the right binaries...

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fee205a..ce7036f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1347,12 +1347,12 @@ add_test(NAME nekoc
 )
 
 add_test(NAME nekoml
-	COMMAND nekoc
+	COMMAND nekoml
 	WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
 )
 
 add_test(NAME nekotools
-	COMMAND nekoc
+	COMMAND nekotools
 	WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
 )
 
